agent/src/com/jonnyzzz/teamcity/virtual/run/RelativePaths.java

```diff
@@ -26,13 +27,23 @@
  */
 public class RelativePaths {
   @NotNull
-  public static String resolveRelativePath(@NotNull final File baseDir, @NotNull final File path) {
-    String result = FileUtil.getRelativePath(baseDir, path);
+  public static String resolveRelativePath(@NotNull final File baseDir, @NotNull final File path, boolean isWindows, boolean convertToLinux) {
+    String result = isWindows
+            ? FileUtil.getRelativePath(baseDir.getAbsolutePath(), path.getAbsolutePath(), '\\')
+            : FileUtil.getRelativePath(baseDir.getAbsolutePath(), path.getAbsolutePath(), '/');
     if (result == null) return "";
-
-    result = result.replace('\\', '/');
-
-    while (result.endsWith(".") || result.endsWith("/")) result = result.substring(0, result.length() - 1);
+    if(convertToLinux)
```

Comment:
It looks like we do need similar conversion for `!linux` case as well (e.g. if the code is running on Windows targeting Linux containers)